src/main/java/app/bpartners/geojobs/service/event/AnnotationBatchRetrievingSubmittedService.java

```diff
@@ -0,0 +1,78 @@
+package app.bpartners.geojobs.service.event;
+
+import static java.time.Instant.now;
+import static java.util.UUID.randomUUID;
+
+import app.bpartners.gen.annotator.endpoint.rest.model.Annotation;
+import app.bpartners.gen.annotator.endpoint.rest.model.AnnotationBatch;
+import app.bpartners.geojobs.endpoint.event.model.AnnotationBatchRetrievingSubmitted;
+import app.bpartners.geojobs.endpoint.rest.model.TileCoordinates;
+import app.bpartners.geojobs.repository.model.detection.HumanDetectedObject;
+import app.bpartners.geojobs.repository.model.detection.HumanDetectedTile;
+import app.bpartners.geojobs.repository.model.tiling.Tile;
+import app.bpartners.geojobs.service.annotator.AnnotationService;
+import app.bpartners.geojobs.service.detection.HumanDetectedTileService;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class AnnotationBatchRetrievingSubmittedService
+    implements Consumer<AnnotationBatchRetrievingSubmitted> {
+  private final AnnotationService annotationService;
+  private final HumanDetectedTileService humanDetectedTileService;
+
+  @Override
+  public void accept(AnnotationBatchRetrievingSubmitted submitted) {
+    var jobId = submitted.getJobId();
+    var annotationJobId = submitted.getAnnotationJobId();
+    var annotationTaskId = submitted.getAnnotationTaskId();
+    var imageSize = submitted.getImageSize();
+    var xTile = submitted.getXTile();
+    var yTile = submitted.getYTile();
+    var zoom = submitted.getZoom();
+    var annotationBatches = annotationService.getAnnotations(annotationJobId, annotationTaskId);
+    var humanDetectedTiles =
```

Comment:
From this line 38 to line 61, refactor the code and put it inside a Consumer class.

Reply:
It cannot be put in Consumer because building a humanDetectedTile requires some extra parameters not only annotationBatches